docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1156 @@
+
+This means that references to :class:`~qiskit.opflow.state_fns.StateFn` in opflow code should be examined to
+identify the sub-class that is being used, to then look for an alternative.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+
+   * - :class:`~qiskit.opflow.state_fns.StateFn`
+     - In most cases, :class:`~qiskit.quantum_info.Statevector`. Remember that this is a factory class.
+
+   * - :class:`~qiskit.opflow.state_fns.CircuitStateFn`
+     - :class:`~qiskit.quantum_info.Statevector`
+
+   * - :class:`~qiskit.opflow.state_fns.DictStateFn`
+     - :class:`~qiskit.quantum_info.Statevector`
+
+   * - :class:`~qiskit.opflow.state_fns.VectorStateFn`
+     - This class can be replaced with :class:`~qiskit.quantum_info.Statevector` or
+       :class:`~qiskit.quantum_info.StabilizerState` (for Clifford-based vectors).
+
+   * - :class:`~qiskit.opflow.state_fns.SparseVectorStateFn`
+     - No direct replacement.
+
+   * - :class:`~qiskit.opflow.state_fns.OperatorStateFn`
+     - :class:`~qiskit.quantum_info.Statevector`
+
+   * - :class:`~qiskit.opflow.state_fns.CVaRMeasurement`
+     - Used in :class:`~qiskit.opflow.expectations.CVaRExpectation`.
+       Functionality now covered by :class:`~SamplingEstimator`. See :ref:`example <cvar>` in expectations.
+
+
+Example 1: Applying an operator to a state
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Opflow</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit.opflow import StateFn, X, Y
+
+    qc = QuantumCircuit(2)
+    op = X ^ Y
+    state = StateFn(qc)
+
+    comp = ~state @ op
+    # returns a CircuitStateFn
+
+    eval = comp.eval()
+    # returns a VectorStateFn (Statevector)
+
+.. raw:: html
+
+   </details>
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Alternative</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit import QuantumCircuit
+    from qiskit.quantum_info import SparsePauliOp, Statevector
+
+    qc = QuantumCircuit(2)
+    op = SparsePauliOp("XY")
+    state = Statevector(qc)
+
+    eval = state.evolve(operator)
+    # returns a Statevector
+
+.. raw:: html
+
+   </details>
+
+
+See more applied examples in :ref:`expectations <expect_state>`  and :ref:`converters <convert_state>`.
+
+
+Converters
+----------
+*Back to* `Index`_
+
+The role of this sub-module was to convert the operators into other opflow operator classes
+(:class:`~qiskit.opflow.converters.TwoQubitReduction`, :class:`~qiskit.opflow.converters.PauliBasisChange`...).
+In the case of the :class:`~qiskit.opflow.converters.CircuitSampler`, it traversed an operator and outputted
+approximations of its state functions using a quantum backend.
+Notably, this functionality has been replaced by the :mod:`~qiskit.primitives`.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+
+   * - :class:`~qiskit.opflow.converters.CircuitSampler`
+     - :class:`~primitives.Estimator`
```

Comment:
I thought the CircuitSampler was more akin to Sampler. It takes States (described by circuits) and executes them to into Dict or Vector states akin to the probabilities. There are the expectations and basis change logic that used in conjunction with this ends up being more akin to Estimator.

Reply:
As the circuit sampler was the base for all circuit evaluations, I'd think that both Sampler and Estimator are a replacement for this.